### ISSUE TITLE
ConcurrentModificationException on getEnabledRuleNames

### ISSUE THREAD

**Issue by Tapchicoma** [facebook/facebook-android-sdk#870]
## Checklist

- [x] I've updated to the latest released version of the SDK
- [x] I've searched for existing [GitHub issues](https://github.com/facebook/facebook-android-sdk/issues)
- [x] I've looked for existing answers on [Stack Overflow](https://facebook.stackoverflow.com), the [Facebook Developer Community Forum](https://developers.facebook.com/community/) and the [Facebook Developers Group](https://www.facebook.com/groups/fbdevelopers)
- [x] I've read the [Code of Conduct](CODE_OF_CONDUCT.md)
- [x] This issue is not security related and can safely be disclosed publicly on GitHub

## Environment

Describe your dev environment here, giving as many details as possible. If you have them, make sure to include:

- Java Version: `1.8.0_256`
- Android SDK Version: `4.1.0`
- Installation Platform & Verison: `[Maven|Gradle|Manual]` version Gradle `6.7`
- Facebook sdk version: `8.1.0`

## Actual Results

After updating to `8.1.0` SDK version and making a new release we start to observing following crashes:
```java
Fatal Exception: java.util.ConcurrentModificationException
       at java.util.HashMap$HashIterator.nextNode(HashMap.java:1441)
       at java.util.HashMap$KeyIterator.next(HashMap.java:1465)
       at com.facebook.appevents.aam.MetadataRule.getEnabledRuleNames(MetadataRule.java:96)
       at com.facebook.appevents.UserDataStore.getEnabledInternalUserData(UserDataStore.java:193)
       at com.facebook.appevents.UserDataStore.getAllHashedUserData(UserDataStore.java:187)
       at com.facebook.internal.Utility.setAppEventAttributionParameters(Utility.java:769)
       at com.facebook.applinks.AppLinkData.fetchDeferredAppLinkFromServer(AppLinkData.java:155)
       at com.facebook.applinks.AppLinkData.access$000(AppLinkData.java:48)
       at com.facebook.applinks.AppLinkData$1.run(AppLinkData.java:143)
       at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
       at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
       at java.lang.Thread.run(Thread.java:919)
```

**Comment by linmx0130** [comment]
Thanks for raising this issue! It seems that commit [253e0cc](https://github.com/facebook/facebook-android-sdk/commit/253e0cc3d9310b254d4f7642cd61a62202d665b3) has fixed this problem and the bug fix was delivered in v8.2.0. Does it still happen?

**Comment by dmo60** [comment]
@linmx0130 The issue is fixed for us with the v.8.2.0 release. Thank you!
